testutil/network.go

```diff
@@ -0,0 +1,368 @@
+			apiAddr = fmt.Sprintf("http://%s:%s", apiURL.Hostname(), apiURL.Port())
+
+			rpcAddr, _, err := server.FreeTCPAddr()
+			require.NoError(t, err)
+			tmCfg.RPC.ListenAddress = rpcAddr
+		}
+
+		logger := log.NewNopLogger()
+		if cfg.EnableLogging {
+			logger = log.NewTMLogger(log.NewSyncWriter(os.Stdout))
+			logger, _ = tmflags.ParseLogLevel("info", logger, tmcfg.DefaultLogLevel())
+		}
+
+		ctx.Logger = logger
+
+		nodeDirName := fmt.Sprintf("node%d", i)
+		nodeDir := filepath.Join(network.BaseDir, nodeDirName, "simd")
+		clientDir := filepath.Join(network.BaseDir, nodeDirName, "simcli")
+		gentxsDir := filepath.Join(network.BaseDir, "gentxs")
+
+		require.NoError(t, os.MkdirAll(filepath.Join(nodeDir, "config"), 0755))
+		require.NoError(t, os.MkdirAll(clientDir, 0755))
+
+		tmCfg.SetRoot(nodeDir)
+		tmCfg.Moniker = nodeDirName
+		monikers[i] = nodeDirName
+
+		proxyAddr, _, err := server.FreeTCPAddr()
+		require.NoError(t, err)
+		tmCfg.ProxyApp = proxyAddr
+
+		p2pAddr, _, err := server.FreeTCPAddr()
+		require.NoError(t, err)
+		tmCfg.P2P.ListenAddress = p2pAddr
+		tmCfg.P2P.AddrBookStrict = false
+		tmCfg.P2P.AllowDuplicateIP = true
+
+		nodeID, pubKey, err := genutil.InitializeNodeValidatorFiles(tmCfg)
+		require.NoError(t, err)
+		nodeIDs[i] = nodeID
+		valPubKeys[i] = pubKey
+
+		kb, err := keyring.New(sdk.KeyringServiceName(), keyring.BackendTest, clientDir, buf)
+		require.NoError(t, err)
+
+		addr, secret, err := server.GenerateSaveCoinKey(kb, nodeDirName, cfg.Passphrase, true)
+		require.NoError(t, err)
+
+		info := map[string]string{"secret": secret}
+		infoBz, err := json.Marshal(info)
+		require.NoError(t, err)
+
+		// save private key seed words
+		require.NoError(t, writeFile(fmt.Sprintf("%v.json", "key_seed"), clientDir, infoBz))
+
+		balances := sdk.NewCoins(
+			sdk.NewCoin(fmt.Sprintf("%stoken", nodeDirName), cfg.AccountTokens),
+			sdk.NewCoin(cfg.BondDenom, cfg.StakingTokens),
+		)
+
+		genFiles = append(genFiles, tmCfg.GenesisFile())
+		genBalances = append(genBalances, banktypes.Balance{Address: addr, Coins: balances.Sort()})
+		genAccounts = append(genAccounts, authtypes.NewBaseAccount(addr, nil, 0, 0))
+
+		createValMsg := stakingtypes.NewMsgCreateValidator(
+			sdk.ValAddress(addr),
+			valPubKeys[i],
+			sdk.NewCoin(sdk.DefaultBondDenom, cfg.BondedTokens),
+			stakingtypes.NewDescription(nodeDirName, "", "", "", ""),
+			stakingtypes.NewCommissionRates(sdk.OneDec(), sdk.OneDec(), sdk.OneDec()),
+			sdk.OneInt(),
+		)
+
+		p2pURL, err := url.Parse(p2pAddr)
+		require.NoError(t, err)
+
+		memo := fmt.Sprintf("%s@%s:%s", nodeIDs[i], p2pURL.Hostname(), p2pURL.Port())
+		tx := authtypes.NewStdTx([]sdk.Msg{createValMsg}, authtypes.StdFee{}, []authtypes.StdSignature{}, memo) //nolint:staticcheck // SA1019: authtypes.StdFee is deprecated
+		txBldr := authtypes.TxBuilder{}.
+			WithChainID(cfg.ChainID).
+			WithMemo(memo).
+			WithKeybase(kb)
+
+		signedTx, err := txBldr.SignStdTx(nodeDirName, tx, false)
```

Comment:
Can we use `TxGenerator`/`tx.Factory` here please since all the required functionality is now merged in?

Reply:
Can you point me to a place in the codebase that I can reference on how to do this?

Reply:
This is what I've attempted to no avail:

```go
		memo := fmt.Sprintf("%s@%s:%s", nodeIDs[i], p2pURL.Hostname(), p2pURL.Port())
		txb := cfg.TxGenerator.NewTxBuilder()
		require.NoError(t, txb.SetMsgs(createValMsg))
		txb.SetMemo(memo)

		txf := tx.Factory{}.WithKeybase(kb).
			WithChainID(cfg.ChainID).
			WithMemo(memo).
			WithSignMode(signing.SignMode_SIGN_MODE_LEGACY_AMINO_JSON).
			WithTxGenerator(cfg.TxGenerator)

		tx.Sign(txf, nodeDirName, txb)

		txBz, err := cfg.TxGenerator.TxEncoder()(txb.GetTx())
		require.NoError(t, err)
```

```
=== RUN   TestIntegrationTestSuite
    TestIntegrationTestSuite: network_test.go:17: setting up integration test suite
    TestIntegrationTestSuite: network.go:161: acquiring test network lock
    TestIntegrationTestSuite: network.go:166: created temporary directory: /var/folders/4q/7kvqvdrx3mj_zq6f7fd7fwrw0000gp/T/chain-t3EnWZ544130682
    TestIntegrationTestSuite: network.go:175: preparing test network...
    TestIntegrationTestSuite: network.go:336: 
        	Error Trace:	network.go:336
        	            				network_test.go:19
        	            				suite.go:118
        	            				network_test.go:37
        	Error:      	Received unexpected error:
        	            	cannot parse disfix JSON wrapper: invalid character '(' looking for beginning of value
        	Test:       	TestIntegrationTestSuite
--- FAIL: TestIntegrationTestSuite (0.15s)
```

Reply:
I really don't feel like looking into investigating and fixing anything in `gentuitil` WRT gentxs. I'm going to leave the code as-is for now.

Reply:
Hmm... that's too bad. Alright, can you just place a TODO comment there and we'll get to it once `genutil` is migrated